`no-undef` false positive

From https://twitter.com/ymyzk/status/703426419481661440 (Japanese)

`no-undef` is warning `new.target` meta property.

```
> eslint -v
v2.2.0
> echo "class A { constructor() { new.target; } }" | eslint --stdin --env es6 --rule no-undef:2 --no-eslintrc

<text>
  1:27  error  'new' is not defined     no-undef
  1:31  error  'target' is not defined  no-undef

✖ 2 problems (2 errors, 0 warnings)
```

This seems to be a bug of `escope`.
`escope` has been creating references of `new` and `target` identifiers.
I will try to fix this.

--- a/package.json
+++ b/package.json
@@ -41,7 +41,7 @@
     "debug": "^2.1.1",
     "doctrine": "^1.2.0",
     "es6-map": "^0.1.3",
-    "escope": "^3.4.0",
+    "escope": "^3.5.0",
     "espree": "^3.1.1",
     "estraverse": "^4.1.1",
     "estraverse-fb": "^1.3.1",

--- a/tests/lib/rules/no-undef.js
+++ b/tests/lib/rules/no-undef.js
@@ -62,6 +62,9 @@ ruleTester.run("no-undef", rule, {
         { code: "/*global b:false*/ var b = 1;" },
         { code: "Array = 1;" },
 
+        // new.target: https://github.com/eslint/eslint/issues/5420
+        { code: "class A { constructor() { new.target; } }", parserOptions: { ecmaVersion: 6 } },
+
         // Experimental,
         {
             code: "var {bacon, ...others} = stuff; foo(others)",

